### PR TITLE
Automatic Screen Tracking

### PR DESCRIPTION
**Pull request by f2prateek**

### CODE REVIEW

File: Pod/Classes/SEGAnalytics.m [hunk 1]
```diff
@@ -5,11 +5,11 @@
 #import "SEGAnalyticsUtils.h"
 #import "SEGAnalyticsRequest.h"
 #import "SEGAnalytics.h"
-
 #import "SEGIntegrationFactory.h"
 #import "SEGIntegration.h"
-#import <objc/runtime.h>
 #import "SEGSegmentIntegrationFactory.h"
+#import "UIViewController+SEGScreen.h"
+#import <objc/runtime.h>
```
Comment (tonyxiao) [inline Pod/Classes/SEGAnalytics.m]: Unneeded import?

File: Pod/Classes/Internal/UIViewController+SEGScreen.m [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                                method_getImplementation(originalMethod),
+                                method_getTypeEncoding(originalMethod));
+        } else {
+            method_exchangeImplementations(originalMethod, swizzledMethod);
+        }
+    });
+}
+
+
++ (UIViewController *)topViewController
```
Comment (tonyxiao) [inline Pod/Classes/Internal/UIViewController+SEGScreen.m]: We should prefix our method names if we were to add category to global classes such as `UIViewController`. something like `seg_topViewController`

File: Pod/Classes/Internal/UIViewController+SEGScreen.m [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        UINavigationController *navigationController = (UINavigationController *)rootViewController.presentedViewController;
+        UIViewController *lastViewController = [[navigationController viewControllers] lastObject];
+        return [self topViewController:lastViewController];
+    }
+
+    UIViewController *presentedViewController = (UIViewController *)rootViewController.presentedViewController;
+    return [self topViewController:presentedViewController];
+}
+
+- (void)screen_viewDidAppear:(BOOL)animated
```
Comment (tonyxiao) [inline Pod/Classes/Internal/UIViewController+SEGScreen.m]: I think should probably use the same consistent prefix as well - `seg_`.

File: Pod/Classes/Internal/UIViewController+SEGScreen.m (hunk 2 above)
Comment (f2prateek) [inline Pod/Classes/Internal/UIViewController+SEGScreen.m]: Ah good point

File: Pod/Classes/Internal/UIViewController+SEGScreen.m (hunk 3 above)
Comment (f2prateek) [inline Pod/Classes/Internal/UIViewController+SEGScreen.m]: Gotcha

File: Pod/Classes/SEGAnalytics.m (hunk 1 above)
Comment (f2prateek) [inline Pod/Classes/SEGAnalytics.m]: Good catch! This was actually in the file from earlier for some reason....

Comment (f2prateek) [inline Pod/Classes/SEGAnalytics.m]: Ah we use it in forwarding events to methods e.g. https://github.com/segmentio/analytics-ios/blob/master/Pod/Classes/SEGAnalytics.m#L553

**Comment by tonyxiao** [comment]
👍 

PS: `xcodeproj` files are so hard to diff.

**Comment by f2prateek** [comment]
@tonyxiao Haha agree, is it safe to simply uncheck them Git? I copied the .gitignore config from somewhere and it didn't have it.

Maybe we could check it in but treat it as a binary file http://shanesbrain.net/2008/7/9/using-xcode-with-git

**Pull request merged**